tools/build_script_generator/scons/resources/SConscript.in

```diff
@@ -55,6 +55,16 @@ env.Append(CPPDEFINES=[
 %# ================================== STM32 ===================================
 %% if platform in ["stm32"]
 env["CPU"] = "{{ core }}"
+%% if fp in ["4f", "7f", "7fd"]
+env["FPU_ABI"] = "-mfloat-abi=softfp"
```

Comment:
Is there a reason not to use `-mfloat-abi=hard`?
https://gcc.gnu.org/onlinedocs/gcc/ARM-Options.html

Reply:
Hm, when I tried some years ago it failed because some library wasn't compiled with the same settings (I think it was newlib), but I can't reproduce this anymore. Let's see if `hard` ABI works in the CI.